src/lib.rs

```diff
@@ -384,6 +388,13 @@ impl Backend {
 
                     scheduler.local(Self::on_proc_macro_error);
                 }
+                recv(analysis_progress_status_receiver) -> analysis_progress_status => {
+                    let Ok(analysis_progress_status) = analysis_progress_status else { break };
+
+                    scheduler.local(|state: &mut State, _notifier: Notifier, requester: &mut Requester, _responder: Responder|
```

Comment:
```suggestion
                    scheduler.local(|state, _notifier, requester, _responder|
```
No need to explicitly type it, inference will do the job here